applet/src/com/redhat/ckey/applet/Unsigned64.java

```diff
@@ -0,0 +1,211 @@
+// the GNU Lesser General Public License Version 2.1 (the "LGPL"), in which
+// case the provisions of the LGPL are applicable instead of those above. If
+// you wish to allow use of your version of this file only under the terms
+// of the LGPL, and not to allow others to use your version of this file
+// under the terms of the BSD license, indicate your decision by deleting
+// the provisions above and replace them with the notice and other
+// provisions required by the LGPL. If you do not delete the provisions
+// above, a recipient may use your version of this file under the terms of
+// either the BSD license or the LGPL.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+// END LICENSE_BLOCK
+
+package com.redhat.ckey.applet;
+
+//import javacard.framework.ISOException;
+//import javacard.framework.JCSystem;
+import javacard.framework.Util;
+
+// Represent and unsigned 64 quantity as a list of 8 short quantities stored in memory.
+// All that is needed is a decrement operation and XOR operation for the AES KWP alg.
+class Unsigned64 {
+	private final short MAX_BYTE_LEN = (short) 16;
+	private final short MAX_SHORT_LEN = (short) 8;
+        //Make public for efficiency copying one object to another
+	public byte[] data = null;
+	public short dataOffset = 0;
+
+        Unsigned64() {
+        }
+
+	public void setData(byte[] input, short inOffset) {
+            data = input;
+	    dataOffset = inOffset;
+	}
+
+        //set the value from an array of bytes
+        public void setFromBytes(byte[] input, short offset) {
+            if(input == null || offset < 0) {
+                return;
+            }
+
+	    short val = 0;
+	    for(short i = 0 ; i < MAX_SHORT_LEN ; i++) {
+                val = Util.makeShort((byte) 0x0, input[(short) (i+offset)]);
+	        setShortAt(i,val);
+	    }
+        }
+
+        // Set the values to 0
+        public void clear() {
+	    Util.arrayFillNonAtomic(data, (short) dataOffset, (short) MAX_BYTE_LEN,(byte) 0);
+        }
+
+	//Get the value as a byte array, being placed into the output buffer.
+        public void getBytes(byte[] output, short offset) {
+            if(output == null || ((short) (offset + MAX_SHORT_LEN)) > output.length) {
+                return;
+            }
+
+	    short bOffset = dataOffset;
+	    short val = 0;
+	    for (short i = 0; i < MAX_SHORT_LEN ; i++) {
+	        val = getShortAt(i);
+		output[(short) (offset + i)] = (byte) val;
+	    }
+        }
+
+	//Set the value based on an input short value
+        public void setFromShort(short in) {
+            if(in < 0) {
+                return;
+            }
+	    clear();
+	    setShortAt((short) 6 ,(short)(in >> MAX_SHORT_LEN));
+	    setShortAt((short) 7, (short) (in & 0x00ff));
```

Comment:
This doesn't look right.  It should be setShortAt((short) 3, in)
or 
setByteAt((short) 6, (short)(in >> MAX_BYTE_LEN))
setByteAt((short)7,(short)(in & 0x00ff));